app/pods/components/goodie-comp/component.js

```diff
@@ -0,0 +1,61 @@
+import Component from '@ember/component';
+import { alias, and, not, equal } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { task } from 'ember-concurrency';
+import DS from 'ember-data';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+    api: service(),
+    router: service(),
+    
+    showModal: false,
+    collapsed: true,
+    tshirt: '',
+    run: alias('runAttempt.run'),
+    goodieRequests: alias('runAttempt.goodieRequests'),
+    statusInProgress: alias('goodieRequests.statusInProgress'),
+    thresholdCompleted: computed('statusInProgress','alreadyClaimed','progress', 'run.goodiesThreshold', function () {
+        const thresholdCompleted = this.progress.then(progress => {
+            return progress.completedContents / progress.totalContents > (this.get('run.goodiesThreshold')/100)
+        })
+        return DS.PromiseObject.create({
+            promise: thresholdCompleted
+        })
+    }),
+    progress: computed('runAttempt', function () {
+        return DS.PromiseObject.create({
+            promise: this.api.request(`run_attempts/${this.runAttempt.id}/progress`)
+        })
+    }),
+
+    canClaim: alias('thresholdCompleted'),
+    alreadyClaimed: equal('statusInProgress', 'requested'),
+    completed: equal('statusInProgress', 'completed'),
+    
+    actions: {
+        toggleCollapse() {
+            this.toggleProperty('collapsed');
+        },
+        toggleModal() {
+            this.toggleProperty('showModal');
+        },
+        saveForm() {
```

Comment:
use ember-concurrency task instead and show ui feedback ( `task-button` component)